### PR TITLE
Intro to exceptions

### PR DESCRIPTION
**Pull request by CheezItMan**
Straightforward conversion of textbook repo.

### CODE REVIEW

File: 03-language-features/exception-handling.md [hunk 1]
```diff
@@ -0,0 +1,464 @@
...
+      return planet if planet.name == name
+    end
+    return nil
+  end
+end
+```
+
+So when are exceptions appropriate? There are a lot of [differing opinions](https://www.sitepoint.com/ruby-error-handling-beyond-basics/), but the consensus seems to be that exception handling should be for *exceptional* circumstances.
+
+Here are some examples of when and when not to raise an exception. Each of the following situations is paired with an example in the context of `GroceryStore`.
```
Comment (beccaelenzil) [inline 03-language-features/exception-handling.md:401]: Is `GroceryStore` still the project for next week?

File: 03-language-features/exception-handling.md [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+```
+
+Like method or block parameters, the rescued exception is a local variable and may be called anything you please.
+
+### Exercise
+
+Setup:
+1. Create a new file called `csv_printer.rb` in your working directory.
+1. Copy the following code into that file.
+1. Review what the code is doing with another student, if you can.
```
[the remark this answers is not in the document]
Comment (kaidamasaki) [reply 03-language-features/exception-handling.md:205]: Yes!

File: 03-language-features/exception-handling.md [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+
+* type: short-answer
+* id: c6f9ef0f-631e-487c-94da-c733186d41b1
+* title: What does ArgumentError.new tell us?
+* points: 1
+* topics: exception-handling
+
+##### !question
+
+We know that raising an `ArgumentError` requires a call to `ArgumentError.new`. What does this tell us about what `ArgumentError` is, and what sort of a thing we're raising?
```
[the remark this answers is not in the document]
Comment (CheezItMan) [reply 03-language-features/exception-handling.md:32]: Yeah, I'll need to rework this.

File: 03-language-features/exception-handling.md (hunk 1 above)
Comment (CheezItMan) [reply 03-language-features/exception-handling.md]: Nope moved to Adagrams

File: 03-language-features/exception-handling.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (CheezItMan) [reply 03-language-features/exception-handling.md]: Done

**Pull request merged**